src/presto/month_date.rs

```diff
@@ -0,0 +1,57 @@
+// start implementing uses
+use arrow::array::ArrayRef;
+use arrow::compute::{cast, date_part, DatePart};
+use arrow::datatypes::DataType;
+use datafusion::common::Result;
+use datafusion::logical_expr::{ColumnarValue, ScalarUDFImpl, Signature, Volatility};
+use std::any::Any;
+
+use crate::utils::make_scalar_function;
+// end implementing uses
+
+#[derive(Debug)]
+pub(super) struct Func {
+    signature: Signature,
+}
+
+impl Func {
+    pub fn new() -> Self {        
+        // start implementing constructor
+        Self {
+            signature: Signature::exact(vec![DataType::Date32], Volatility::Immutable),
+        }
+        // end implementing constructor
+    }
+}
+
+impl ScalarUDFImpl for Func {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+    fn name(&self) -> &str {
+        "month"
+    }
+
+    fn signature(&self) -> &Signature {
+        &self.signature
+    }
+
+    // start implementing return_type
+    fn return_type(&self, _arg_types: &[DataType]) -> Result<DataType> {
+        Ok(DataType::Int64)
+    }
+    // end implementing return_type
+
+    // start implementing invoke
+    fn invoke(&self, args: &[ColumnarValue]) -> Result<ColumnarValue> {
+        make_scalar_function(month, vec![])(args)
```

Comment:
OMG... no, we should absolutely not be calling `make_scalar_function` from inside of `invoke`! This was the old deprecated adaptor for lazy scalar functions, it does an Arc allocation, it was only meant to called *once per function*, not *once per function call*!